File: C++/Practice/OPPS/CopyConstructor.cpp
#include <iostream>
using namespace std;

class code
{
    int id;
    string name;

    public :
    code(){}
    code(int a){id = a;name ="Ramesh";}
    code(code &x)
    {
        id = x.id;
        name ="Ramesh";
    }
    void display(void)
    {
        cout << id << endl;
        cout << name << endl;
    }
};

int main()
{
    code A(100);
    code B(A);
    code C = A;

    code D;
    D = A;

    cout << "\nid of A : " ; A.display();
    cout << "\nid of B : " ; B.display();
    cout << "\nid of C : " ; C.display();
    cout << "\nid of D : " ; D.display();
    
    return 0;
}

File: Java/Constructorr/CopyConstructor.java
package Java.Constructorr;

import java.util.Scanner;

public class CopyConstructor { // Name should be Student
    int id;
    String name;
    CopyConstructor(int i,String n)
    {
        id = i;
        name = n;
    }
    CopyConstructor(CopyConstructor c1)
    {
        id=c1.id;
        name=c1.name;
    }
    void display()
    {
        System.out.println(id+" "+name);
    }
    public static void main(String[] args) {
        Scanner sc  = new Scanner(System.in);
        CopyConstructor c1 = new CopyConstructor(11,"Ramesh");
        CopyConstructor c2 = new CopyConstructor(c1);
        c1.display();
        c2.display();
        sc.close();
    }
}
